giskard/scanner/scanlogger.py

```diff
@@ -0,0 +1,65 @@
+import enum
+from typing import Optional
+
+from giskard.scanner.xprint import (
+    BLACK_STYLE,
+    BLUE_STYLE,
+    BOLD,
+    CHARS_LIMIT,
+    CYAN_STYLE,
+    GREEN_STYLE,
+    MAGENTA_STYLE,
+    RED_STYLE,
+    WHITE_STYLE,
+    YELLOW_STYLE,
+    Template,
+    xprint,
+)
+    
+class ScanLoggerLevel(enum.Enum):
+    DEBUG = 0
+    INFO = 5
+    CRITICAL = 10
+
+class ScanLogger:
+    level: ScanLoggerLevel = ScanLoggerLevel.INFO
+    
+    def setLevel(self, 
+        level: ScanLoggerLevel
+    ):
+        if not isinstance(level, ScanLoggerLevel):
+            raise ValueError("Invalid log level. Please provide a ScanLoggerLevel enum.")
```

Comment:
you can simply do instead:
```py
@configured_validate_arguments
def setLevel(self, level: ScanLoggerLevel):
    self.level = level
```

see https://github.com/Giskard-AI/giskard/blob/e13588dedbee34e8a661ad1ce797a6e2d40de5b0/giskard/core/validation.py#L16

Reply:
I changed the code so that it matches strings to ScanLoggerLevel objects. It looks like this:

```
def setLevel(self, level: str):
        level_map = {
            "DEBUG": ScanLoggerLevel.DEBUG,
            "INFO": ScanLoggerLevel.INFO,
            "CRITICAL": ScanLoggerLevel.CRITICAL
        }

        if level not in level_map:
            raise ValueError(f"Invalid log level: {level}. Please provide 'DEBUG', 'INFO', or 'CRITICAL'.")
        
        self.level = level_map[level]
```